### ISSUE TITLE
In case of no custom header set, the items selected header does not show up on selecting check boxes

### ISSUE THREAD

**Issue by bhavyalatha26** [maxim-saplin/data_table_2#92]
Hi @maxim-saplin,

First of all thank you for this wonderful package. I have been playing around with this package. And wanted to suggest a change for the table header section.

At the moment, I did not set any `header` and when I try selecting check boxes, the `n items selected` header does not show up. 

```
AsyncPaginatedDataTable2(
            controller: _controller,
            columns: _cols,
            // header: Text("My Header"),
            ...
            ... 
)           
```

![image](https://user-images.githubusercontent.com/13586777/169681253-1e905c0e-e372-48b5-ae58-510e1d4174d6.png)

But if I set some `header` , then the items selected tile shows up.
```
AsyncPaginatedDataTable2(
            controller: _controller,
            columns: _cols,
            header: Text("My Header"),
            ...
            ... 
)           
```
![image](https://user-images.githubusercontent.com/13586777/169681274-197823ab-73ec-49a5-895a-554ea208b713.png)
![image](https://user-images.githubusercontent.com/13586777/169681285-17aa4afd-48b0-4b13-aaf8-58656b3cac74.png)

I went through the source code, and I think in the `_getHeader()` function as defined in [paginated_data_table_2.dart#L584](https://github.com/maxim-saplin/data_table_2/blob/19ba24f653b55efa0a40357ba41a2bbd8e6fe3d5/lib/src/paginated_data_table_2.dart#L584) we need to handle also the case `if (_selectedRowCount > 0)`  and just add the `selectedRowCountTitle` to header widgets.

Regards,
Bhavya.

**Comment by maxim-saplin** [comment]
Hi @bhavyalatha26 . You suggest to override the standard **header** behavior of `PaginatedDataTable2` which always shows the number of selected items. Am I right?

**Comment by bhavyalatha26** [comment]
Yes @maxim-saplin, even if no header is set by the user, the selected items header should show up whenever something is selected

**Comment by maxim-saplin** [comment]
OK, I had different unerstanding. My original assumption was that if there's a header widget, then developers should have the capability to change what is displayed in header's place whenever items are selected.

They way I get you last comment is thay you would like to be able to see header with the number of selected items even if the `header` widget is not provided. In such a case I don't see why not use empty widget in `header` param, e.g. `SizedBox()` and get empty space in teh header and get the selected summary when selectioj is made

**Comment by bhavyalatha26** [comment]
Ya @maxim-saplin I already tried this workaround, but the problem is it shows empty white space, that looks unwanted in the UI. I tried to use `SizedBox.shrink()` to keep it negligible but it takes a fixed height (not sure where is this height getting fixed) thus shows unwanted empty white space above the table. 

```
AsyncPaginatedDataTable2(
            controller: _controller,
            columns: _cols,
            header: SizedBox.shrink(),
            ...
            ... 
) 
```

> Doesn't matter what's the header widget (for ex : even for `SizedBox(height: 1000, width: 1000,)`), it always takes fixed height.

![image](https://user-images.githubusercontent.com/13586777/169754398-974d4b99-da2b-4ed8-aa2e-e4c734e21135.png)


Only when I do not set the header, the unwanted space is gone.

```
AsyncPaginatedDataTable2(
            controller: _controller,
            columns: _cols,
            //header: SizedBox.shrink(),
            ...
            ... 
) 
```
![image](https://user-images.githubusercontent.com/13586777/169754944-b93f803a-6c5f-4f9e-9cff-aa14532c9bac.png)

**Comment by maxim-saplin** [comment]
If there's no empty space the table will jump upon each selection. That's a very distracting user experience, are you sure you'd like that behaviour?

**Comment by maxim-saplin** [comment]
For the moment I don't see here a workable feature that requires changing of the widget iteself. If you still want the behaviour you described I'd suggest creating a custom widget atop of data table and implement the required behaviour - shouldn't be hard

**Comment by bhavyalatha26** [comment]
Hi @maxim-saplin , ya I tried with my custom widget too, but the point I got stuck is, how to fetch the **count of the selected rows** outside the `AsyncPaginatedDataTable2` widget ?

For instance I tried to add a container on top of the table with a text field like below

```
Container(
          height: 50,
          width: double.infinity,
          color: Colors.black87,
          padding: EdgeInsets.symmetric(horizontal: 40.0),
          alignment: Alignment.centerLeft,
          child: Text(
            "${_dessertsDataSource!.selectedRowCount} rows selected",
            style: context.bodyMedium!.copyWith(color: Colors.white),
          ),
        )
 ```

![image](https://user-images.githubusercontent.com/13586777/169939767-1f576c18-d419-4212-9e84-cee442a0786b.png)
 
But, when I select any rows the changes reflect inside the Datasource class but not outside in the above container wdiget.

**2 rows selected**
![image](https://user-images.githubusercontent.com/13586777/169939940-45556b2c-6bf4-43e9-9a20-bffe5e16eabf.png)

**Logs (inside data source class):**
```
DataTable2 built: 0ms
Selected rows: 0
DataTable2 built: 0ms
Selected rows:  1
DataTable2 built: 0ms
Selected rows :  2
DataTable2 built: 0ms
```


If you can suggest me some approach to resolve this, it would be helpful.

Regards,
Bhavya.

**Comment by maxim-saplin** [comment]
_dessertsDataSource - it contains rows and their selected state, that's where the DataTable2 takes the info if a row is selected

**Comment by maxim-saplin** [comment]
Though your contrainer is not updated whenever dataSource is updated. I'd suggest changin the data source class and adding there an callback, e.g. OnSelectedCountChanged and calling it whenever a row is selected. Please note that at data source there's `getRow` method whcih creates data rows and each row has `onSelectChanged` event - that would be the right place to add the call to `OnSelectedCountChanged`

**Comment by bhavyalatha26** [comment]
> Though your contrainer is not updated whenever dataSource is updated. I'd suggest changin the data source class and adding there an callback, e.g. OnSelectedCountChanged and calling it whenever a row is selected. Please note that at data source there's `getRow` method whcih creates data rows and each row has `onSelectChanged` event - that would be the right place to add the call to `OnSelectedCountChanged`

Thanks for the tip. I get it to some extent, can you just help with  a sample code snippet on how `onSelectedCountChanged` function should look like.

**Comment by bhavyalatha26** [comment]
Also to give you a heads up, my container is able to get count when I select all check boxes, but the count does not reflect on individual selection of check boxes.

![image](https://user-images.githubusercontent.com/13586777/170108743-b6b598ad-45c6-4721-b691-15a386799a7d.png)

![image](https://user-images.githubusercontent.com/13586777/170108864-91b82919-0ab5-4dd5-a398-cee4c4d4f3b1.png)

This is how my data row looks like:

````
return AsyncRowsResponse(
          data['totalRows'],
          (data['rows'] as List<dynamic>).map((contactJson) {
            var contact =
                CompanyContact.fromJson(contactJson as Map<String, dynamic>);
            return DataRow(
              key: ValueKey<int>(contact.id ?? 0),
              selected: false,
              onSelectChanged: (value) {
                if (value != null) {
                  setRowSelection(ValueKey<int>(contact.id ?? 0), value);
                  print("Selected $selectedRowCount rows");
                }
              },
              cells: [
                DataCell(Text(contact.firstName ?? "")),
                DataCell(Text(contact.lastName ?? "")),
                DataCell(Text(contact.companyName ?? "")),
                DataCell(Text(contact.phone ?? "")),
              ],
            );
          }).toList());
````

Am I missing anything here ?

**Comment by bhavyalatha26** [comment]
Ok I finally go to make it work by adding a listener in the widget class that defines my data source. Thanks for the tip about callback

```
MyDataSourceAsync? _dessertsDataSource;

@override
  void initState() {
    _dessertsDataSource ??= MyDataSourceAsync();
    _dessertsDataSource!.addListener(() {
      if (_dessertsDataSource!.selectedRowCount >= 0){
        setState(() {
        });
      }
    });
    super.initState();
  }
  ```

**Comment by maxim-saplin** [comment]
Great you found the solution!
